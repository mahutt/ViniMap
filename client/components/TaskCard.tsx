import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

interface TaskCardProps {
  text: string;
  onDelete: () => void;
}

const TaskCard = ({ text, onDelete }: TaskCardProps) => {
  return (
    <View style={styles.item}>
      <View style={styles.itemLeft}>
        <View style={styles.square}></View>
        <Text style={styles.itemText}>{text}</Text>
      </View>
      <TouchableOpacity style={styles.circular} onPress={onDelete}>
        <Icon name="close" size={18} color="white" />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    backgroundColor: '#FFF',
    padding: 15,
    borderRadius: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 20,
    elevation: 2,
  },
  itemLeft: {
    flexDirection: 'row',
    alignItems: 'center',
    flexWrap: 'wrap',
  },
  square: {
    width: 24,
    height: 24,
    backgroundColor: '#852C3A',
    opacity: 0.4,
    borderRadius: 5,
    marginRight: 15,
  },
  itemText: {
    maxWidth: '80%',
  },
  circular: {
    width: 20,
    height: 20,
    backgroundColor: '#852C3A',
    borderRadius: 18,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 3,
  },
});

export default TaskCard;
