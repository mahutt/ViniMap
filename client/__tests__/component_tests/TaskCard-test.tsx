import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import TaskCard from '@/components/TaskCard';
import { OpaqueColorValue, StyleProp, TextStyle } from 'react-native';

type IconSymbolProps = {
  name: string;
  size?: number;
  color: string | OpaqueColorValue;
  style?: StyleProp<TextStyle>;
};

jest.mock('@/components/ui/IconSymbol', () => ({
  IconSymbol: (props: IconSymbolProps) => {
    return <React.Fragment>IconSymbol</React.Fragment>;
  },
}));

describe('TaskCard', () => {
  const mockProps = {
    text: 'Sample Task',
    selected: false,
    onDelete: jest.fn(),
    onSelect: jest.fn(),
    modifyTask: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    const { toJSON } = render(<TaskCard {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the task text correctly', () => {
    const { getByText } = render(<TaskCard {...mockProps} />);
    expect(getByText('Sample Task')).toBeTruthy();
  });

  it('applies selected styles when task is selected', () => {
    const selectedProps = { ...mockProps, selected: true };
    const { toJSON } = render(<TaskCard {...selectedProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onSelect when the checkbox is pressed', () => {
    const { UNSAFE_getAllByProps } = render(<TaskCard {...mockProps} />);

    const touchables = UNSAFE_getAllByProps({ activeOpacity: 0.7 });
    fireEvent.press(touchables[0]);
    expect(mockProps.onSelect).toHaveBeenCalledTimes(1);
  });

  it('calls modifyTask when the edit button is pressed', () => {
    const { UNSAFE_getAllByProps } = render(<TaskCard {...mockProps} />);

    const iconButtons = UNSAFE_getAllByProps({
      style: expect.objectContaining({ backgroundColor: '#852C3A' }),
    });

    fireEvent.press(iconButtons[0]);
    expect(mockProps.modifyTask).toHaveBeenCalledTimes(1);
  });

  it('calls onDelete when the delete button is pressed', () => {
    const { UNSAFE_getAllByProps } = render(<TaskCard {...mockProps} />);

    const iconButtons = UNSAFE_getAllByProps({
      style: expect.objectContaining({ backgroundColor: '#852C3A' }),
    });

    fireEvent.press(iconButtons[1]);
    expect(mockProps.onDelete).toHaveBeenCalledTimes(1);
  });
});
