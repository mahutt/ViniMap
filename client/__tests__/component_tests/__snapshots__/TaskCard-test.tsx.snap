// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TaskCard correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 10,
        "elevation": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 15,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#852C3A",
              "borderRadius": 5,
              "height": 24,
              "marginRight": 15,
              "opacity": 0.4,
              "width": 24,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#000",
            "maxWidth": "80%",
          }
        }
      >
        Test Task
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#852C3A",
          "borderRadius": 18,
          "elevation": 3,
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 20,
        }
      }
    >
      <ViewManagerAdapter_SymbolModule
        animated={false}
        colors={[]}
        name="xmark"
        resizeMode="scaleAspectFit"
        style={
          [
            {
              "height": 24,
              "width": 24,
            },
            [
              {
                "height": 14,
                "width": 14,
              },
              undefined,
            ],
          ]
        }
        tintColor="white"
        type="monochrome"
        weight="regular"
      />
    </View>
  </View>
</View>
`;
